easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -537,29 +538,43 @@ def include_defined_parameters(keyset):
                         else:
                             val = newval
 
-                        ebtxt.append("%s = %s" % (key, val))
+                        add_key_and_comments(key, val)
+
                         printed_keys.append(key)
                         printed = True
                 if printed:
                     ebtxt.append('')
 
+        def add_key_and_comments(key, val):
+            """ Adds key, value pair and comments (if there are any) to the dump file """
+            if key in self.comments['inline']:
```

Comment:
I think it's ugly to use the self from the outer scope,
can't you make this a private method instead of an inner method, and pass self along?